overlord/snapstate/snapstate.go

```diff
@@ -2145,6 +2260,64 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []mi
 	return updated, updateTss, nil
 }
 
+type updateGroup struct {
+	infos []minimalInstallInfo
+	names []string
+}
```

Comment:
Maybe one info+name could have their own struct and updateGroup be a slice of them

Reply:
This was a not very elegant way of keeping track of infos/names in the same group before passing them into doUpdate. Having just one info+names in the struct meant I would have to rebuild each list before passing to doUpdate. I've had a better look and I think it's fine to actually pass all the names into each call of doUpdate because the extra ones will be ignored. I left a comment explaining why in case this looks odd to someone. This simplifies this logic quite a bit